runner-macros/src/entry.rs

```diff
@@ -0,0 +1,148 @@
+    args: syn::AttributeArgs,
+    is_test: bool,
+) -> Result<TokenStream, syn::Error> {
+    if input.sig.asyncness.take().is_none() {
+        let msg = "the `async` keyword is missing from the function declaration";
+        return Err(syn::Error::new_spanned(input.sig.fn_token, msg));
+    }
+
+    let mut flavor = None;
+    for arg in args {
+        match arg {
+            syn::NestedMeta::Meta(syn::Meta::NameValue(namevalue)) => {
+                let ident = namevalue
+                    .path
+                    .get_ident()
+                    .ok_or_else(|| {
+                        syn::Error::new_spanned(&namevalue, "Must have specified ident")
+                    })?
+                    .to_string()
+                    .to_lowercase();
+
+                let msg = format!("Not expected to received a value for: {}", ident.as_str());
+                return Err(syn::Error::new_spanned(namevalue, msg));
+            }
+            syn::NestedMeta::Meta(syn::Meta::Path(path)) => {
+                let name = path
+                    .get_ident()
+                    .ok_or_else(|| syn::Error::new_spanned(&path, "Must have specified ident"))?
+                    .to_string()
+                    .to_lowercase();
+                match name.as_str() {
+                    "sandbox" => {
+                        flavor = Some(Flavor::Sandbox);
+                    }
+                    "testnet" => {
+                        flavor = Some(Flavor::Testnet);
+                    }
+                    "mainnet" => {
+                        flavor = Some(Flavor::Mainnet);
+                    }
+                    name => {
+                        let msg = format!("Unknown attribute {} is specified; expected one of: `sandbox`, `testnet`, `mainnet`", name);
+                        return Err(syn::Error::new_spanned(path, msg));
+                    }
+                };
+            }
+            other => {
+                return Err(syn::Error::new_spanned(
+                    other,
+                    "Unknown attribute inside the macro",
+                ));
+            }
+        }
+    }
+
+    // If type mismatch occurs, the current rustc points to the last statement.
+    let (last_stmt_start_span, last_stmt_end_span) = {
+        let mut last_stmt = input
+            .block
+            .stmts
+            .last()
+            .map(ToTokens::into_token_stream)
+            .unwrap_or_default()
+            .into_iter();
+        // `Span` on stable Rust has a limitation that only points to the first
+        // token, not the whole tokens. We can work around this limitation by
+        // using the first/last span of the tokens like
+        // `syn::Error::new_spanned` does.
+        let start = last_stmt.next().map_or_else(Span::call_site, |t| t.span());
+        let end = last_stmt.last().map_or(start, |t| t.span());
+        (start, end)
+    };
+
+    let rt = match flavor.unwrap() {
```

Comment:
Perhaps this should also allow configuring through env variable? Can just open an issue to tackle later, but would be much more preferable to just set an env variable to override where the tests are run instead of forcing code changes to test against diff networks.

runner-js has `NEAR_RUNNER_NETWORK`

Reply:
👍 Will make that an issue to tackle later